epoxy-adapter/src/main/java/com/airbnb/epoxy/BaseEpoxyAdapter.java

```diff
@@ -140,6 +138,32 @@ protected void onModelBound(EpoxyViewHolder holder, EpoxyModel<?> model, int pos
 
   }
 
+  @CallSuper
+  @Override
+  public void onAttachedToRecyclerView(RecyclerView recycler) {
+    super.onAttachedToRecyclerView(recycler);
+
+    // set our item animator decorator so we can intercept calls to the item animator
+    // and the user doesn't have to worry about setting the decorator himself
+    if (!(recycler.getItemAnimator() instanceof ChangeAnimationDelegator)) {
```

Comment:
> If their custom animator is for insertions/removals and they don't want to do anything for item changes, then would we still run that reflection code in the delegate animator?

yes

Since there would be problems when setting an animator after the adapter, we can't go with the `onAttachedToRecyclerView` approach anyway, so the user will have to set the Decorator themself and thus has complete control over when it's used. 

We can also subclass `DefaultItemAnimator` to avoid reflection when the user wants to use only `DefaultItemAnimator` and not a custom `ItemAnimator`. There should definitely be a way to use any `ItemAnimator` with our decorator though.

We should profile many adds/removes/changes and see if reflection could become a performance problem (what a good opportunity to try the new profiler in AS). I don't think the reflective listener is called when scrolling (except at the end of endless RVs)